lib/preview/link.js

```diff
@@ -102,18 +102,41 @@ var getYoutubeId = function(link) {
  */
 var processYoutube = function(ctx, youtubeId, callback) {
   var thumbnailUrl = util.format('http://img.youtube.com/vi/%s/hqdefault.jpg', youtubeId);
-  CoreUtil.download(ctx, thumbnailUrl, 'thumbnail.jpq', function(err, thumbnailPath, mimeType) {
+  CoreUtil.download(ctx, thumbnailUrl, 'youtube_thumbnail.jpg', function(err, thumbnailDownloadPath, mimeType) {
     if (err) {
       return callback(err);
     } else if (mimeType !== 'image/jpeg') {
-      return callback({'code': 500, 'msg': 'Unexpected mime type for a youtube thumbnail'});
+      return callback({'code': 500, 'url': thumbnailUrl, 'msg': 'Unexpected mime type for a YouTube thumbnail image'});
     }
 
-    var metadata = {
-      'youtubeId': youtubeId
-    };
-    var result = new Result(PreviewConstants.STATUS.DONE, thumbnailPath, null, null, metadata);
-    return callback(null, result);
+    PreviewImage.generateThumbnail(ctx, thumbnailDownloadPath, function(err, thumbnailPath) {
+      if (err) {
+        return callback(err);
+      }
+
+      var previewUrl = util.format('http://img.youtube.com/vi/%s/maxresdefault.jpg', youtubeId);
```

Comment:
You're right, @johncrossman - seems like hqdefault.jpg is the largest image that's guaranteed to be there. Amended and testing on dev; will signal once it's checked out.